scripts/compress_images_test.py

```diff
@@ -0,0 +1,238 @@
+
+        self._create_test_image(self.png_path, 'PNG')
+        self._create_test_image(self.jpg_path, 'JPEG')
+        self._create_test_image(self.webp_path, 'WEBP')
+
+    def tearDown(self) -> None:
+        """Clean up test environment after each test."""
+        shutil.rmtree(self.temp_dir)
+
+    def _create_test_image(self, path: str, Img_format: str) -> None:
+        """Create a test image for compression testing."""
+        img = Image.new('RGB', (100, 100), color='red')
+        img.save(path, format=Img_format)
+
+    def test_get_compression_type_zip_extension(self) -> None:
+        """Test get_compression_type returns 'Zip' for PNG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.png')
+        self.assertEqual(compression_type, 'Zip')
+
+    def test_get_compression_type_lzw_extension(self) -> None:
+        """Test get_compression_type returns 'LZW' for JPG."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        compression_type = compressor.get_compression_type('.jpg')
+        self.assertEqual(compression_type, 'LZW')
+
+    def test_compress_images(self) -> None:
+        """Test image compression process."""
+        with self.swap(subprocess, 'run', mock.MagicMock()):
+            self.assertFalse(os.path.exists(self.output_dir))
+            result_images: List[CompressedImageInfo] = [
+                {
+                    'path': pathlib.Path(self.png_path),
+                    'original_size': 1000,
+                    'new_size': 500
+                }
+            ]
+            compressor = compress_images.ImageCompressor(
+                self.temp_dir,
+                output_dir=self.output_dir
+            )
+            compressor.compress_images(result_images)
+
+            self.assertTrue(os.path.exists(self.output_dir))
+
+            rel_path = os.path.relpath(self.png_path)
+            output_file_path = pathlib.Path(self.output_dir) / rel_path
+            self.assertTrue(output_file_path.exists())
+
+            if pathlib.Path(self.png_path).exists():
+                self.assertEqual(
+                    output_file_path.stat().st_size,
+                    pathlib.Path(self.png_path).stat().st_size
+                )
+
+    def test_compress_images_directory_removal_error(self) -> None:
+        """Test error handling when removing output directory fails."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+        mock_rmtree = mock.MagicMock(ignore_errors=OSError('Permission denied'))
+        mock_log_error = mock.MagicMock()
+        mock_subprocess_run = mock.MagicMock(
+            return_value=mock.MagicMock(
+                returncode=1,
+                stderr='Permission denied error'
+            )
+        )
```

Comment:
here and elsewhere, please either create your mock objects from scratch or, if you want to use `mock.Mock` or similar tools, declare which attributes the mocks should have (and declare the minimal set of attributes possible) so that accessing anything else raises an error. At the end of the day, we want our tests to ensure that anything we mock is being used as expected by the code under test